### PR TITLE
Add some performance metrics to sql_bank

### PR DESCRIPTION
**Pull request by vivekmenezes**

### CODE REVIEW

File: examples/sql_bank/main.go [hunk 1]
```diff
@@ -58,21 +60,28 @@ UPDATE bank.accounts
...
+			if err != nil {
 				if log.V(1) {
 					log.Warning(err)
 				}
 				continue
 			}
-			// TODO(pmattis): We should only be updating numTransfers when rows were
-			// actually updated. See #2377.
-			atomic.AddUint64(&numTransfers, 1)
+			if result == driver.ResultNoRows {
```
Comment (tamird) [inline examples/sql_bank/main.go]: how could this ever be the case? ResultNoRows is only returned on DDLs

File: examples/sql_bank/main.go [hunk 2]
```diff
@@ -58,21 +60,28 @@ UPDATE bank.accounts
...
+				continue
+			}
+			// Read and write in the same statement.
+			reading.read = time.Since(start)
+			reading.write = reading.read
+			readings <- reading
```
Comment (tamird) [inline examples/sql_bank/main.go]: i think i'd prefer a struct literal here

File: examples/sql_bank/main.go [hunk 3]
```diff
@@ -37,20 +37,22 @@ import (
...
-func moveMoney(db *sql.DB) {
+func moveMoney(db *sql.DB, readings chan measurement) {
 	for {
 		from, to := rand.Intn(*numAccounts), rand.Intn(*numAccounts)
 		if from == to {
 			continue
 		}
 		amount := rand.Intn(*maxTransfer)
-
+		var reading measurement
```
Comment (tamird) [inline examples/sql_bank/main.go]: can we make this as narrowly scoped as possible?

File: examples/sql_bank/main.go [hunk 4]
```diff
@@ -202,20 +206,28 @@ func main() {
...
 
 	for range time.NewTicker(*balanceCheckInterval).C {
 		now := time.Now()
 		elapsed := time.Since(lastNow)
-		numTransfers := atomic.LoadUint64(&numTransfers)
-		log.Infof("%d transfers were executed at %.1f/second.", (numTransfers - lastNumTransfers), float64(numTransfers-lastNumTransfers)/elapsed.Seconds())
-		verifyBank(db)
-		lastNumTransfers = numTransfers
 		lastNow = now
+		transfers := len(readings)
```
Comment (tamird) [inline examples/sql_bank/main.go]: https://gobyexample.com/range-over-channels ?

Comment (petermattis) [inline examples/sql_bank/main.go]: Not sure that can be used here. Range over channel will block until the channel is closed.

Comment (tamird) [inline examples/sql_bank/main.go]: good point.

**Comment by petermattis** [comment]
We should move (in a separate PR) `{sql,kv}_bank` into the examples-go repo.

File: examples/sql_bank/main.go (hunk 1 above)
Comment (vivekmenezes) [inline examples/sql_bank/main.go]: put back the TODO

File: examples/sql_bank/main.go (hunk 3 above)
Comment (vivekmenezes) [inline examples/sql_bank/main.go]: I've converted all measurements to struct literals.

File: examples/sql_bank/main.go [hunk 5]
```diff
@@ -37,20 +36,21 @@ import (
...
 var numAccounts = flag.Int("num-accounts", 999, "Number of accounts.")
 var concurrency = flag.Int("concurrency", 5, "Number of concurrent actors moving money.")
-var transferStyle = flag.String("transfer-style", "single-stmt", "\"single-stmt\" or \"txn\"")
+var transferStyle = flag.String("transfer-style", "txn", "\"single-stmt\" or \"txn\"")
 var usePostgres = flag.Bool("use-postgres", false, "Use postgres instead of cockroach.")
 var balanceCheckInterval = flag.Duration("balance-check-interval", 1*time.Second, "Interval of balance check.")
 
-var numTransfers uint64
+type measurement struct {
+	read, write, total time.Duration
```
Comment (vivekmenezes) [inline examples/sql_bank/main.go]: I added total to this change

File: examples/sql_bank/main.go (hunk 1 above)
Comment (tamird) [inline examples/sql_bank/main.go]: I think fixing the TODO is good, but the right way to do it is by calling result.RowsAffected

**Comment by tamird** [comment]
LGTM

**Comment by petermattis** [comment]
LGTM

File: examples/sql_bank/main.go (hunk 1 above)
Comment (vivekmenezes) [inline examples/sql_bank/main.go]: On Thu, Oct 1, 2015 at 4:50 PM Tamir Duberstein <notifications@github.com>
wrote:

> In examples/sql_bank/main.go
> <https://github.com/cockroachdb/cockroach/pull/2743#discussion_r40964566>:
>
> >  				if log.V(1) {
> >  					log.Warning(err)
> >  				}
> >  				continue
> >  			}
> > -			// TODO(pmattis): We should only be updating numTransfers when rows were
> > -			// actually updated. See #2377.
> > -			atomic.AddUint64(&numTransfers, 1)
> > +			if result == driver.ResultNoRows {
>
> I think fixing the TODO is good, but the right way to do it is by calling
> result.RowsAffected
>
Sounds good! Done

File: examples/sql_bank/main.go [hunk 6]
```diff
@@ -37,20 +36,21 @@ import (
...
+}
 
-func moveMoney(db *sql.DB) {
+func moveMoney(db *sql.DB, readings chan measurement) {
```
Comment (tschottdorf) [inline examples/sql_bank/main.go]: you could pass `readings <-chan measurement`.

**Comment by tschottdorf** [comment]
LGTM

**Pull request merged**